src/components/description-list/_macro.njk

```diff
@@ -1,23 +1,34 @@
 {% macro onsDescriptionList(params) %}
     <dl
-        class="ons-description-list ons-description-list__items ons-grid ons-grid--gutterless ons-u-cf{{ " " + params.classes if params.classes else "" }}"
-        {% if params.id %}id="{{ params.id }}"{% endif %}{% if params.descriptionListLabel %}
+        class="ons-description-list ons-description-list__items{{ " " + params.classes if params.classes else "" }}{{ " ons-description-list--horizontal" if params.horizontal else " ons-grid ons-grid--gutterless ons-u-cf" }}"
```

Comment:
I think the only class that needs to be removed here when you are using horizontal is `ons-u-cf`